cmd/asset-syncer/utils.go

```diff
@@ -358,17 +361,83 @@ func extractFilesFromBuffer(buf *bytes.Buffer) (*artifactFiles, error) {
 
 // Charts retrieve the list of charts exposed in the repo
 func (r *OCIRegistry) Charts() ([]models.Chart, error) {
-	// TBD
-	return []models.Chart{}, nil
+	result := []models.Chart{}
+	url, err := parseRepoURL(r.RepoInternal.URL)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, appName := range r.repositories {
+		chartInfo := models.Chart{}
+		firstTag := true
+		for _, tag := range r.tags[appName].Tags {
+			// Pull Chart
+			// TODO: Run in parallel
+			ref := path.Join(url.Host, url.Path, fmt.Sprintf("%s:%s", appName, tag))
+			chartBuffer, digest, err := r.puller.pullOCIChart(ref)
+			if err != nil {
+				return nil, err
+			}
+
+			// Extract
+			files, err := extractFilesFromBuffer(chartBuffer)
```

Comment:
So we're loading the files for every tag into memory here I think. Is that going to be a problem if there's many tags for an app? (or do we implicitly do this already with the current non-oci?)

Reply:
we are extracting one tag at a time, it should not be a problem

Reply:
> we are extracting one tag at a time,

Not sure what you mean? The code above iterates all the tags for the app and appears to extract the file contents into memory for each tag (ie. could be 200 copies of the same/similar readme, icons etc. in memory). Probably not that big, but I'm just not sure if we were doing this before or explicitly appending the file data later (one at a time).

Or how do you mean that it currently does them one at a time (as in, writes the file and frees the memory before getting the next tag)? 

> it should not be a problem

Hopefully not :)

Reply:
sorry, yes, I was thinking just about this `files` var but it's true that we are storing all these files in `ChartVersions` for later. Taking Jenkins as an example, the three files there (readme, values, schema) (note that the icons are not fetched here), are ~60KB. The problem is that this is per version. If we have 10 versions it would be 600KB. And then if we have 200 apps the count goes up to 120MB, it can really be a problem for large repositories.

Just thinking out loud here but maybe it's better to just store the pointer to the `Reader` to those files so we can read and store the files just when needed. Will evaluate that.